modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridMapQueryExecutor.java

```diff
@@ -998,6 +999,25 @@ private GridQueryNextPageResponse prepareNextPage(
         }
     }
 
+    /** */
+    private static List<Value[]> toPlainRows(List<Value[]> rows, int cols) {
+        if (F.isEmpty(rows) || rows.get(0).length == cols)
```

Comment:
What if lengths of other rows are greater then `cols`?

Reply:
I believe there are no such case.
Do you have some specific scenario in mind?